packages/cli/README.md

```diff
@@ -1,66 +1,478 @@
+
+```javascript
+// ./src/cli/commands/build.js
+
+export const handler = async ({
+  side = ['api', 'web'],
+  verbose = false,
+  stats = false,
+}) => {
+  ...
+}
+```
+
+The logic that goes in a command's `handler`'s varies too much to comment on generally. But you'll see similarities among the handlers of commands that do similar things, like generators. 
+
+<!-- ### builder and testing
+
+While we want you to export `builder` as a function, you can use a `builderObject` to define the ags/options.
+
+We recommend doing this esp. for testing purposes, since you can import your builderObject in tests and use it to mock an object...
+
+export const builder = (yargs) => {
+  positional...
+} -->
+
+### Adding an Entry Point Command
+
+If you're adding a command that serves as an entry point to more commands, like `db`, `destroy`, and `generate`, you'll want to create
+1) a file for the command in `./src/commands`, like in [Adding a Command](#adding-a-command), and
+2) a directory to store all the commands it serves as an entry point to.
+
+Using the `generate` command as an example, in `./src/commands`, there's the file `generate.js` and the directory `generate`.
+
+> Again, although you don't have to name the file and the directory after the command, it's best practice to do so.
+
+Files for entry-point commands typically aren't too complicated. Here's the contents of `generate.js` in its entirety:
+
+```javascript
+export const command = 'generate <type>'
+export const aliases = ['g']
+export const description = 'Save time by generating boilerplate code'
+import terminalLink from 'terminal-link'
+
+export const builder = (yargs) =>
+  yargs
+    .commandDir('./generate', { recurse: true })
+    .demandCommand()
+    .epilogue(
+      `Also see the ${terminalLink(
+        'Redwood CLI Reference',
+        'https://redwoodjs.com/reference/command-line-interface#generate-alias-g'
+      )}`
+    )
+
+```
+
+The call to `commandDir` is what makes this command an entry point. You can read more about `commandDir` [here](https://github.com/yargs/yargs/blob/master/docs/advanced.md#commanddirdirectory-opts).
+
+> The second argument to `commandDir`, `{ recurse: true }` is specific to the `generate` command because each of the commands in the `generate` directory is in its own directory to keep things organized--more on this in [Adding a Generator](#adding-a-generator).
+
+Now all the commands in the `generate` directory will be arguments to the `generate` command:
+
+```
+./src/commands/generate
+├── auth
+├── cell
+├── component
+├── function
+├── helpers.js
+├── layout
+├── page
+├── README.md
+├── scaffold
+├── sdl
+├── service
+└── __tests__
+```
+
+There's files and directories here that aren't yargs related&mdash;`README.md`, `helper.js`, and `__tests__`, but because yargs will only use the files that export the appropriate constants, it's ok.
```

Comment:
```suggestion
There are files and directories here that aren't yargs related&mdash;`README.md`, `helper.js`, and `__tests__`, but because yargs will only use the files that export the appropriate constants, that's ok.
```